modules/andes-core/broker/src/main/java/org/dna/mqtt/moquette/messaging/spi/impl/ProtocolProcessor.java

```diff
@@ -833,6 +833,9 @@ void processSubscribe(ServerChannel session, SubscribeMessage msg, String client
         }
 
         MQTTAuthorizationSubject authSubject = authSubjects.get(clientID);
+        if (authSubject == null) {
```

Comment:
when a client (version less than 3.1.1) is connected to the broker and when it tries to subscribe to a topic which is not authorized to, then the server will disconnect the client forcefully. However if the client does not acknowledge the message then it will try to subscribe back to the topic which would lead to a null pointer exception. 
Need to send a failure ack and have fixed it.